src/ec/eddsa.rs

```diff
@@ -196,8 +198,12 @@ struct Point {
 type Elem = [i32; ELEM_LIMBS];
 const ELEM_LIMBS: usize = 10;
 
-type PublicKey = [u8; ELEM_LEN];
+type PublicKey = [u8; PUBLIC_KEY_LEN];
+const PUBLIC_KEY_LEN: usize = ELEM_LEN;
 
+const PRIVATE_KEY_LEN: usize = ELEM_LEN;
```

Comment:
Okay, looking at it further, there seems to be a little bit of conceptual muddiness in the naming.  For example, consider the signature:
```rust
GFp_x25519_ge_scalarmult_base(h: &mut ExtPoint, a: &Seed)
```
It seems like the second argument here should actually be a `&Scalar`, based on the function name, and what it does (confirmed by the C comment). The only place `Seed` or the above function are used is in the module-private `public_from_private` helper, which in its name and usage accepts the bytes of a private key, but uses a different (more generic) type alias (`Seed`) for its first parameter.

Given that, I think we should get rid of `Seed`, add a `PrivateKey` type alias, and define `PUBLIC_KEY_LEN = 32`, since it isn't _really_ a scalar itself— it is just a bitstring which determines a scalar, and is determined by the Ed25519 `b` parameter, which we never explicitly mention in this code. What do you think?